lib/index.js

```diff
@@ -149,7 +151,7 @@ function getOptions(options) {
 
   options.error = function(err) {
     if (error) {
-      error(util._extend(new Error(), JSON.parse(err)));
+      error.call(options.context, util._extend(new Error(), JSON.parse(err)));
```

Comment:
Sure.

I'm worried about the case where we write this as `importer.call(options, ...)`. For this, options such as `file` will be accessible as `this.file`. It will work as long both libsass and node-sass do not need to share any additional information with custom importers and functions.

In ruby-sass which libsass seeks parity with, there is a concept of a global environment. I can see us providing this in a more node-like way as opposed to the ruby pattern of creating a global. If we are currently binding to `options`, then we would be unable to add this environment information without a breaking change (as references to `this.file` would suddenly fail). The `context` object fulfills this role, allowing us to later add environment information `{ options: options, env: getEnvironment() }` without breaking anyone's code.